core/lib/config/src/configs/en_config.rs

```diff
@@ -0,0 +1,40 @@
+use std::num::{NonZeroU32, NonZeroU64, NonZeroUsize};
+
+use serde::Deserialize;
+use zksync_basic_types::{
+    commitment::L1BatchCommitmentMode, url::SensitiveUrl, L1ChainId, L2ChainId,
+};
+
+#[derive(Debug, Clone, PartialEq, Deserialize)]
+pub struct Pruning {
+    pub enabled: bool,
+    pub chunk_size: Option<u32>,
+    pub removal_delay_sec: Option<NonZeroU64>,
+    pub data_retention_sec: Option<u64>,
+}
+
+#[derive(Debug, Clone, PartialEq, Deserialize)]
+pub struct SnapshotRecovery {
+    pub enabled: bool,
+    pub postgres_max_concurrency: Option<NonZeroUsize>,
+    pub tree_chunk_size: Option<u64>,
+}
+
+/// Temporary config for initializing external node, will be completely replaced by consensus config later
+#[derive(Debug, Clone, PartialEq, Deserialize)]
+pub struct ENConfig {
+    // Genesis
+    pub l2_chain_id: L2ChainId,
+    pub l1_chain_id: L1ChainId,
+    pub l1_batch_commit_data_generator_mode: L1BatchCommitmentMode,
+
+    // Main node configuration
+    pub main_node_url: SensitiveUrl,
+    pub main_node_rate_limit_rps: Option<NonZeroUsize>,
+
+    // En specific components
```

Comment:
Ironically, none of the following args are specific to EN.

- `commitment_generator_max_parallelism` can be readily used on the main node
- `tree_api_remote_url` is already used on the main node as `Web3JsonRpcConfig.tree_api_url` (so it can be removed, right?)
- Snapshot recovery and pruning are not sufficiently mature to use for the main node, but theoretically, they can be used for it.

Reply:
moved it to general or removed, thanks for the clarification